src/main/resources/hudson/plugins/sauce_ondemand/PluginImpl/config.jelly

```diff
@@ -12,9 +13,70 @@
             <f:entry title="${%Override Sauce Connect Path}" field="sauceConnectDirectory">
                 <f:textbox id="directoryBox"/>
             </f:entry>
-            <f:entry title="${%Sauce Connect Options}" field="sauceConnectOptions">
+            <hr/>
+            <div id="validationMessage" style="color: red;"></div>
+            <div id="deprecationMessage" style="color: orange;"></div>
+
+    <script>
+        document.addEventListener('DOMContentLoaded', function () {
+            const options = document.querySelector('input[name="_.sauceConnectOptions"]');
+            const sauceConnectCLIOptions = document.querySelector('input[name="_.sauceConnectCLIOptions"]');
+            const validationMessage = document.getElementById('validationMessage');
+            const deprecationMessage = document.getElementById('deprecationMessage');
+            const saveButton = document.querySelector('.jenkins-button[name="Submit"]');
+            const applyButton = document.querySelector('.jenkins-button[name="Apply"]');
+
+            // Validation function
+            function validateFields() {
+                const optionsValue = options.value.trim();
+                const sauceConnectCLIOptionsValue = sauceConnectCLIOptions.value.trim();
+                let isValid = true;
+                let message = '';
+
+                // Validation logic: Fail if both fields are set
+                if (optionsValue &amp;&amp; sauceConnectCLIOptionsValue) {
+                    isValid = false;
+                    message = '"Sauce Connect Legacy Options" and "Sauce Connect Options" cannot both be set. Please use SC5 style CLI flags only.';
+                }
+
+                // Update UI
+                validationMessage.textContent = message;
+                if (isValid) {
+                    saveButton.removeAttribute('disabled');
+                    applyButton.removeAttribute('disabled');
+                } else {
+                    saveButton.setAttribute('disabled', 'disabled');
+                    applyButton.setAttribute('disabled', 'disabled');
+                }
+
+                // Add a warning for SC4 upgrades
+                let isDeprecated = false;
+                let deprecationText = '';
+
+                if (optionsValue) {
+                    isDeprecated = true;
+                    deprecationText = "Legacy SC4 CLI commands are deprecated. Please update to SC5 style CLI flags."
```

Comment:
Use field names, eg.
"Sauce Connect Legacy Options" is deprecated. Please use "Sauce Connect Options".

Reply:
Fixed in 9fc7c48